src/software/backend/input/replay_logging/replay_reader.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include <experimental/filesystem>
+
+#include "software/proto/sensor_msg.pb.h"
+#include "software/proto/tbots_replay.pb.h"
+
+class ReplayReader
+{
+   public:
+    explicit ReplayReader(const std::string& _replay_dir);
+    std::optional<SensorMsg> getNextMsg();
+
+   private:
+    bool hasNextMsg() const;
+    void nextChunk();
+    size_t max_chunk_idx;
+    size_t cur_chunk_idx;
+    int cur_msg_idx;
```

Comment:
can this be unsigned?

Reply:
that value is only used for indexing into a protobuf repeated field, which uses `int` for indexing.